src/document/common/common-definitions.xsl

```diff
@@ -131,8 +131,9 @@
         <xsl:variable name="definition" as="element()">
             <xsl:apply-templates select="." mode="find-definition"/>
         </xsl:variable>
+        <xsl:variable name="is-deprecated" select="exists($definition/@deprecated)"/>
         <div class="model-entry definition { name() }">
-            <div class="{ if ($is-inline) then 'instance-header' else 'definition-header' }">
+            <div class="{ if ($is-inline) then 'instance-header' else 'definition-header' }{ if ($is-deprecated) then ' deprecated-header' else '' }">
```

Comment:
One slick way to do this would be (without the variable)

```
<div class="... { $definition/@deprecated/' deprecated-header' }"> ...
```

or to be extra defensive (only marking the class if `@deprecated` has some kind of meaningful value):

<div class="... { $definition/@deprecated[matches(.,'\S')]/' deprecated header' }"> ...

I.e. the string value is at the end of a path, which comes up with an empty sequence if no attribute is found.

(But this is only one of many ways to do this.)